tests/scenario/test_setup.py

```diff
@@ -5,15 +5,13 @@
 
 from unittest.mock import MagicMock, PropertyMock, patch
 
-from charm import _TRAEFIK_SERVICE_NAME, TraefikIngressCharm
+from charm import _TRAEFIK_SERVICE_NAME
 from scenario import Container, State
 from scenario.runtime import trigger
 
 
-@patch("charm.KubernetesServicePatch")
-@patch("lightkube.core.client.GenericSyncClient")
 @patch("charm.TraefikIngressCharm.external_host", PropertyMock(return_value="foo.bar"))
-def test_start_traefik_is_not_running(*_):
+def test_start_traefik_is_not_running(*_, traefik_charm):
```

Comment:
Where is the `traefik_charm` variable coming from?

Reply:
[conftest.py](https://stackoverflow.com/questions/13641973/how-and-where-does-py-test-find-fixtures)